haa binary downloader

wayn3 opened the pull request:
Hey RS,

would you be interested in including this binary haa_downloader.sh bash script in the haa repository?

wayn3 committed:
Create haa_downloader.sh

I created this script for downloading the binary files for use with custom update server but would you be interested in including this in the haa repository?

execute (./haa_downloader.sh) to download the latest version into a <version> subfolder of the relative root and copy to the relative root dir
execute with version param (./haa_downlaoder <version>) to download specific version to <version> subfolder (without copying to relative root)

wayn3 committed:
Update haa_downloader.sh

wayn3 committed:
Update haa_downloader.sh

RavenSystem commented:
You must include a LICENSE header, or terms of use... into your file.

RavenSystem commented on haa_downloader.sh:
```diff
@@ -0,0 +1,48 @@
+                do
+                        file_name="$(basename $i)"
+                        echo -n "Saving ./$version/$file_name ... "
+                        wget -qO "$version/$file_name" "$i"
```
Better use curl instead wget

wayn3 replied:
Started with just wget calls to the latest links before I decided to add in other stuff, I’ll update it to use curl commands and include something in the header

wayn3 replied:
replaced wget with curl,
reduced github api calls to 2 and included a count
included a gpl license statement in the code and printed to terminal

wayn3 committed:
Update haa_downloader.sh

inlcuded a gpl license statement
replaced wget with curl
reduced number of ratelimited github calls to 2 per execution
included print out of ratelimit and remaining calls
